R/annual_pd_change_company_technology.R

```diff
@@ -0,0 +1,71 @@
+#' Display the annual pd changes per company and technology.
+#'
+#' @param data A data.frame that contains Merton output data.
+#' @param shock_year Numeric. Vector of length 1 that specifies the year of the
+#'   policy shock.
+#' @param company_filter Character. A vector of companies to display.
+#' @param geography_filter Character. A vector containing the geographical
+#'   specification of the scenario
+#'
+#' @family qa graphing functions
+#'
+#' @return ggplot object
+#'
+#' @export
+annual_pd_change_company_technology <- function(data,
+                                                shock_year = NULL,
+                                                company_filter = NULL,
+                                                geography_filter = NULL) {
+  force(data)
+
+  shock_year %||% stop("Must provide input for 'shock_year'", call. = FALSE)
+  company_filter %||% stop("Must provide input for 'company_filter'", call. = FALSE)
+  geography_filter %||% stop("Must provide input for 'geography_filter'", call. = FALSE)
+
+  data_has_expected_columns <- all(c(
+    "scenario_name", "company_name", "ald_sector", "technology",
+    "scenario_geography", "year", "PD_change"
+  )
+  %in% colnames(data))
+  stopifnot(data_has_expected_columns)
+
+  shock_year <- stringr::str_c("Carbon balance ", shock_year)
+
+  plot <- data %>%
+    dplyr::filter(
+      .data$company_name %in% .env$company_filter &
+        .data$scenario_name == .env$shock_year &
+        .data$scenario_geography %in% geography_filter
+    ) %>%
```

Comment:
Style: With `filter()` it's rare to use `&` and more common to use `, `.  With `subset()` you are forced to use `&`.